src/engine/guess_toolset.bat

```diff
@@ -32,6 +32,21 @@ if NOT "_%VS150COMNTOOLS%_" == "__" (
     set "BOOST_JAM_TOOLSET=vc141"
     set "BOOST_JAM_TOOLSET_ROOT=%VS150COMNTOOLS%..\..\VC\"
     goto :eof)
+
+set VSWHERE_PRE=Microsoft.VisualStudio.Product.
+set VSWHERE_IDS=%VSWHERE_PRE%Community %pre%Professional %pre%Enterprise %pre%BuildTools
+set VSWHERE_REQ=-requires Microsoft.VisualStudio.Component.VC.Tools.x86.x64
+set VSWHERE_PRP=-property installationPath
+set VSWHERE="%ProgramFiles(x86)%\Microsoft Visual Studio\Installer\vswhere.exe"
+if not exist %VSWHERE% set VSWHERE="%ProgramFiles%\Microsoft Visual Studio\Installer\vswhere.exe"
+if not exist %VSWHERE% goto :skip_vswhere
+for /f "usebackq tokens=*" %%i in (`%VSWHERE% -latest -products %VSWHERE_IDS% %VSWHERE_REQ% %VSWHERE_PRP%`) do (
```

Comment:
Thanks for ping. I think better use `-products *` instead of IDS and add version here `-version "[15.0,16.0)"`

`%VSWHERE% -latest -version "[15.0,16.0)" -products * %VSWHERE_REQ% %VSWHERE_PRP%`